valheim-updater

```diff
@@ -5,39 +5,81 @@ CONFIG_DIRECTORY_PERMISSIONS=${CONFIG_DIRECTORY_PERMISSIONS:-755}
 WORLDS_DIRECTORY_PERMISSIONS=${WORLDS_DIRECTORY_PERMISSIONS:-755}
 WORLDS_FILE_PERMISSIONS=${WORLDS_FILE_PERMISSIONS:-644}
 UPDATE_INTERVAL=${UPDATE_INTERVAL:-900}
-STEAMCMD_ARGS=${STEAMCMD_ARGS-validate}
+STEAMCMD_ARGS=${STEAMCMD_ARGS}
+FORCE_VALIDATE=${FORCE_VALIDATE:-false}
+VALHEIM_PLUS_ENABLED=${VALHEIM_PLUS_ENABLED:-true}
 just_started=true
 next_update=$(date +%s)
 
 cd /opt/steamcmd
 main() {
     while :; do
         update_permissions
-        update
+        update_valheim
+        if [ $VALHEIM_PLUS_ENABLED = true ]; then
+            update_valheim_plus
+        fi
         next_update=$(($(date +%s)+$UPDATE_INTERVAL))
         while [ $(date +%s) -lt $next_update ]; do
             sleep 5
         done
     done
 }
 
-update() {
-    local logfile="$(mktemp)"
-    echo "Updating/Validating Valheim Server"
-    ./steamcmd.sh +login anonymous +force_install_dir /opt/valheim_dl +app_update 896660 $STEAMCMD_ARGS +quit
-    rsync -a --itemize-changes --delete --exclude server_exit.drp --exclude steamapps /opt/valheim_dl/ /opt/valheim | tee "$logfile"
-    grep '^[*>]' "$logfile" > /dev/null 2>&1
-    if [ $? -eq 0 ]; then
-        echo "Valheim Server was updated - restarting"
-        supervisorctl restart valheim-server
+install_valheim_plus() {
+    echo "Installing/Updating Valheim Plus"
+    wget -q -O /opt/valheimplus/UnixServer.zip https://github.com/nxPublic/ValheimPlus/releases/latest/download/UnixServer.zip && \
+    unzip -qo /opt/valheimplus/UnixServer.zip -x changelog.txt run_bepinex.sh -d /opt/valheimplus
+    if [ -f /opt/valheim/BepInEx/config/valheim_plus.cfg ]; then
+        cp -a /opt/valheim/BepInEx/config/valheim_plus.cfg /opt/valheimplus/BepInEx/config/valheim_plus.cfg
+    fi
+    chmod 666 /opt/valheimplus/BepInEx/config/valheim_plus.cfg
+    rsync -a --itemize-changes --exclude UnixServer.zip /opt/valheimplus/ /opt/valheim
+}
+
+update_valheim_plus() {
+    remote_md5="$(curl -sL https://github.com/nxPublic/ValheimPlus/releases/latest/download/UnixServer.zip | md5sum | cut -d ' ' -f 1)"
```

Comment:
We can add `jq` and then the download URL size and update time can be retrieved using e.g.
```
curl -sL https://api.github.com/repos/nxPublic/ValheimPlus/releases/latest | jq -r '.assets[] | select(.name == "UnixServer.zip") | .browser_download_url,.size,.updated_at'
```
We can store the resulting update date together with the ZIP, for instance in a UnixServer.zip.updated_at file or something like that. Then on update check we just do that one API call instead of downloading the ZIP file and compare date and size.

Reply:
Ah, I completely forgot about checking against Github's API. Yeah, your suggestion sounds better to me